pkg/core/resources/model/resource.go

```diff
@@ -128,6 +128,13 @@ func Validate(resource Resource) error {
 	return nil
 }
 
+func CheckDeprecated(resource Resource) []string {
```

Comment:
nit: For me `check` should result in true/false or error/not error. I'd name it maybe `Deprecations()`